### ISSUE TITLE
simplify resolution matrix input

### ISSUE THREAD

**Issue by sbailey** [desihub/desispec#21]
io.read_frame() currently returns the sparse resolution matrix data, requiring the user to convert this into an actual matrix for use (via resolution.Resolution(), which replaces the deprecated io.frame.resolution_data_to_sparse_matrix() ).  Some codes even do this multiple times in different places for the same fiber.

Should io.read_frame() just directly convert the 3D resolution matrix data[fiber, diag, wave] into a list of Resolution objects since that is what the user needs in the end?  Is there a reason to keep it as the 3D data?

**Comment by sbailey** [comment]
I got half way through implementing this and now think it is a bad idea:
  * There were several places in the code that actually were using the 3D resolution matrix data as-is; giving them real sparse matrices forced them to back-convert
  * David Kirkby's Resolution() wrapper makes it easier for people to convert to the sparse matrix resolution matrix format anyway: R = Resolution(resolution_data[i])
  * It started to feel complicated to require functions to accept arrays of sparse matrices instead of just a 3D normal array.

Bottom line: I'm closing this ticket as won't fix.  We can revisit this in the future if the use of 3D sparse resolution matrix data is a pain.
